src/CommunityToolkit.Maui.Core/Views/DrawingView/PlatformView/MauiDrawingViewExtensions.shared.cs

```diff
@@ -0,0 +1,205 @@
+using System.Collections.ObjectModel;
+using CommunityToolkit.Maui.Core.Views;
+
+namespace CommunityToolkit.Maui.Core.Extensions;
+
+/// <summary>
+/// Extension methods to support <see cref="IDrawingView"/>
+/// </summary>
+public static class MauiDrawingViewExtensions
+{
+	/// <summary>
+	/// Get smoothed path.
+	/// </summary>
+	public static ObservableCollection<PointF> SmoothedPathWithGranularity(this IEnumerable<PointF> currentPoints, int granularity)
+	{
+		var currentPointsCopy = new List<PointF>(currentPoints);
```

Comment:
If this is a ObservableCollection we don't need to allocate another collection at the end
```suggestion
		var currentPointsCopy = new ObservableCollection<PointF>(currentPoints);
```